pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -80,28 +82,42 @@ func New(_ runtime.Object, h framework.Handle) (framework.Plugin, error) {
 
 // calculatePriority returns the priority of a node. Given the sumScores of requested images on the node, the node's
 // priority is obtained by scaling the maximum priority value with a ratio proportional to the sumScores.
-func calculatePriority(sumScores int64, numContainers int) int64 {
-	maxThreshold := maxContainerThreshold * int64(numContainers)
+func calculatePriority(sumScores, numContainers int64) int64 {
+	maxThreshold := maxContainerThreshold * numContainers
 	if sumScores < minThreshold {
 		sumScores = minThreshold
 	} else if sumScores > maxThreshold {
 		sumScores = maxThreshold
 	}
 
-	return int64(framework.MaxNodeScore) * (sumScores - minThreshold) / (maxThreshold - minThreshold)
+	return framework.MaxNodeScore * (sumScores - minThreshold) / (maxThreshold - minThreshold)
 }
 
 // sumImageScores returns the sum of image scores of all the containers that are already on the node.
 // Each image receives a raw score of its size, scaled by scaledImageScore. The raw scores are later used to calculate
-// the final score. Note that the init containers are not considered for it's rare for users to deploy huge init containers.
-func sumImageScores(nodeInfo *framework.NodeInfo, containers []v1.Container, totalNumNodes int) int64 {
+// the final score. Note that the init containers are also considered and duplicate images are removed when counting the requested images.
+func sumImageScores(nodeInfo *framework.NodeInfo, pod *v1.Pod, totalNumNodes int) (int64, int64) {
 	var sum int64
-	for _, container := range containers {
+	scoredImages := sets.New[string]()
```

Comment:
I am not sure this is worth it, our profiles constantly show various overheads of memory management overheads... Also, would the outcome really be different when the images are deduped?

Reply:
oh.., we're summing the existence image size, do we have any concern here @xiaomudk otherwise I agree with remove this.

Reply:
I thought about it carefully, and the repeated image should also be counted, 
because the ImageLocality plugin The value of is to make the pod start faster, and duplicate images can get higher scores. Start faster.   @denkensk WDYT?

Reply:
Let's think about it and if I am wrong please correct me. 

What affects pod startup time? In ImageLocality plugin, the size of image we need to download will affect, right? So we need to calculate the size of the image we need to download. 

Here we use the existing image size to calculate the size of the image we need to download. The higher the existing image size, the less we need to download right?

If we calculate the duplicate images here, it may not. Let me give you an extreme example, if the pods has 10 containers, 9 need image1(20m), 1 need image2(100m), and  we have two nodes: image1 in node1, image2 in node2, if we calculate the duplicate images here, we will pick node1 right?  But actually in node1, we need to download 100m, but in node2 we just need to download 20m.

Reply:
@kerthcet @ahg-g As denkensk said, do you have any differing opinions?

Reply:
My concern is more about adding logic to handle a corner case, but will add overhead to all other pods. Did you see the case you are referring to in practice?

Reply:
@ahg-g I haven't come across it in practice, and the pod images are always different. So, your suggestion is that we don't need to consider duplicate images?

Reply:
Then I think we should just leave it be.